src/optimizer/operators.py

```diff
@@ -76,13 +89,23 @@ def __eq__(self, other):
             is_subtree_equal = is_subtree_equal and (child1 == child2)
         return is_subtree_equal
 
+    def is_logical(self):
+        return self._opr_type < OperatorType.LOGICALDELIMITER
```

Comment:
Why other two are not considered as logical?

Reply:
It's a mistake when merging the master branch. Fixed.